app/knights/knight.py

```diff
@@ -0,0 +1,35 @@
+from .equipment import Armour, Weapon, Potion
+
+
+class Knight:
+    """Represents a knight with stats and equipment."""
+
+    def __init__(
+        self, name: str, power: int, hp: int, armour: str, weapon: str,
+        potion: str = None
+    ) -> None:
+        self.name = name
+        self.base_power = power
+        self.hp = hp
+        self.armour = [Armour(**a) for a in armour]
```

Comment:
The `armour` parameter is expected to be a list of dictionaries. Ensure that the input data is structured this way to correctly initialize `Armour` instances.